src/Microsoft.Identity.Client/ClientAssertion.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Type containing an assertion representing a clients's credentials. This type is used to allow specific claims to be added to the authentication request.<c>UserAssertion</c>
+    /// See https://aka.ms/msal-net-client-credentials
+    /// </summary>
+    public sealed class ClientAssertion
```

Comment:
The design of this class is very similar to the design of ClientCredential, which we have deprecated in MSAL3. The builder API allows for direct configuration of properties, so instead of 

```csharp
// Old MSAL v2 api
.WithClientCredential(new ClientCredential(secret));

// New MSAL 3 API
.WithClientSecret(string);
```

Reply:
Yes. I should follow up with @henrik-me and @jmprieur to see if we want to alter the original design of this api.

Reply:
I like @bgavrilMS's proposal. See an alternate proposal in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1188#issuecomment-501185871